javagen/src/main/java/com/azure/autorest/template/ModelTemplate.java

```diff
@@ -554,6 +573,163 @@ private void addModelConstructor(ClientModel model, JavaSettings settings, JavaC
                         String.format("%1$s = %2$s;", constantProperty.getName(), constantProperty.getDefaultValue()));
                 }
             });
+        } else if (model.isStronglyTypedHeader() && settings.isCustomStronglyTypedHeaderDeserializationUsed()) {
+            classBlock.lineComment("HttpHeaders containing the raw property values.");
+            classBlock.javadocComment(settings.getMaximumJavadocCommentWidth(), (comment) ->
+                comment.description(String.format("Creates an instance of %1$s class.", model.getName())));
+            classBlock.publicConstructor(String.format("%s(HttpHeaders rawHeaders)", model.getName()), constructor -> {
+                for (ClientModelProperty property : model.getProperties()) {
+                    addConstructorCustomDeserialization(property, constructor);
+                }
+            });
+        }
```

Comment:
Is this ctor affect whether it is eager deserialization or lazy deserialization?
Is the difference of behavior in azure-core?

Ask because I didn't find other difference in generated code on `model.isStronglyTypedHeader() && settings.isCustomStronglyTypedHeaderDeserializationUsed()`

Reply:
OK, see the azure-core PR.